### DISCUSSION TITLE
Allow Containers to be duplicate of Active prefixes

### DISCUSSION THREAD

**Discussion by PieterL75** [netbox-community/netbox#6627] [Ideas]
We're automating our documentation of Azure/AWS used subnets in netbox.
So far it all went fine, when we worked in the Global vrf.

I'm moving some things to a 'vrf' in netbox, and I want to keep the checkbox 'Enforce Unique Space' on that VRF.
What I notice now, is that for some setups, the 'vnet' and the 'subnets' are the same.
![image](https://user-images.githubusercontent.com/74899468/122541969-809e6480-d02a-11eb-8088-a53f3f2d2ae7.png)

So we make a container per VNETand then assign subnets to it using 'active' prefixes.
But in the above case, the subnet in the VNET takes the entire container, thus creating a 'duplicate' prefix in the vrf.

Would it be an option to create a second checkbox in the vrf that ignores containers for duplicate checks with 'active' prefixes ?
Maybe that containers are only check with containers, and active with active/reserved/deprecated ?

Pieter

**Comment by candlerb** [comment]
For this case, where the active prefix uses the *whole* address space, I think you don't need a container: in Netbox its only purpose is to reserve space for other prefixes to be allocated from and grouped together.  If you want to allocate the whole prefix, you just flip it from container to active.

I do see where you're coming from though.  It's not wrong in principle to allow an entire container pool to be carved out by a single prefix which uses all the space.  However it would add a lot of complexity to both the uniqueness checking and the GUI.  Consider that if you have two prefixes 10.150.36.0/24 - which one should be shown first?  Is one a "child" of the other?  I suspect you would like to consider the container as "parent" and the active as "child", but since the two prefixes are identical, no such relationship is currently inferred.

**Comment by michaelklose** [comment]
Have you found a solution? I have the same issue with Azure vnet and subnet.

Michael

**Reply by PieterL75** [reply]
We unchecked the box that forces a vrf to contain unique prefixes...
I actually completely forgot about this issue. We still need to find a proper implementation with the DevOps team
